### PR TITLE
Update structure & Improvements to introduction

### PR DESCRIPTION
**Pull request by manoranjith**
- Update the structure of documentation so that state-channel section
  appears as a sub-section under concepts section.

- Improve the content in the project introduction and perun-node
  introduction sections.

### CODE REVIEW

File: source/introduction.rst [hunk 1]
```diff
@@ -106,7 +112,7 @@ A multi-user node that uses the go-perun SDK to run the Perun protocol and
 provides an interface for users to manage their keys/identities; off-chain
 networking; open, transact and settle state-channels.
 
-For detailed information on perun-node, see here.
+For detailed information on perun-node, read the :ref:`Introduction:Perun-node` section.
```
Comment (anagha-ks) [inline source/introduction.rst:115]: link not working

File: source/introduction.rst [hunk 2]
```diff
@@ -26,18 +27,21 @@ Perun Protocol
...
-Perun’s state-channel virtualization
+and the University of Warsaw. Perun protocol allows users to shift
+transaction and smart contract execution away from the blockchain into
+so-called payment and state-channels. These channels are created by locking
+coins on the blockchain and can be updated directly between the users and
+without any on-chain interaction. This makes state-channel-based transactions
+much faster and cheaper than on-chain transactions. The underlying blockchain
+guarantees that all off-chain transactions will be enforced on-chain
+eventually. For a detailed explanation on the concept of state channels,
+please read section on :ref:`State Channels`.
```
Comment (anagha-ks) [inline source/introduction.rst:38]: link not working

**Review by anagha-ks — CHANGES_REQUESTED** [review]

File: source/introduction.rst (hunk 2 above)
Comment (manoranjith) [reply source/introduction.rst:38]: I am not sure if the link will work on github, but on my local machine I tried and it works. Can you provide more info here on when it does not work ? @anagha-ks

File: source/introduction.rst (hunk 1 above)
Comment (manoranjith) [reply source/introduction.rst]: Fixed.

File: source/introduction.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (manoranjith) [reply source/introduction.rst:38]: Duplicate of comment below, hence resolving it.

**Review by anagha-ks — APPROVED** [review]

File: source/introduction.rst (hunk 2 above)
Comment (anagha-ks) [reply source/introduction.rst:38]: Yea, this will not work on github, but works on sphinx page.

**Pull request merged**